client/app/models/milestone.js

```diff
@@ -137,15 +137,29 @@ export default class MilestoneModel extends Model {
       displayDate = this.dcpActualstartdate;
     }
 
-    if (this.dcpMilestone === FILED_EAS_REVIEW) { // FILED_EAS_REVIEW is used here as the generic "EAS" milestone
-      // 2792: if the project starts with P, prefare the display date found in the "PREPARE_FIELD_EAS" milestone
-      if (this.project.get('dcpName').startsWith('P')) {
-        const prepareFiledEas = this.project.get('milestones').findBy('dcpMilestone', PREPARE_FILED_EAS);
-        if (prepareFiledEas) {
-          displayDate = prepareFiledEas.dcpActualenddate;
+    if ([ //  Task 2792 - "Environmental Assessment Statement Filed" is displayed
+      PREPARE_FILED_EAS,
+      FILED_EAS_REVIEW,
+    ].includes(this.dcpMilestone)) {
+      if (this.project.get('dcpName').startsWith('P')) { // projectID starts with "P" (so use the “Prepare Filed EAS” milestone)
+        if (this.dcpMilestone === PREPARE_FILED_EAS) { // if it's already a Prepare Filed EAS, use its actual completion date
```

Comment:
nit: can you change the comment here to 'use its actual end date' to be consistent with what you wrote on 147? I wasn't sure if they were two different fields at first.